ot/mpz-ot-core/src/ferret/mpcot/receiver_regular.rs

```diff
@@ -0,0 +1,192 @@
+//! MPCOT receiver for regular indices. Regular indices means the indices are evenly distributed.
+
+use mpz_core::Block;
+
+use crate::ferret::mpcot::error::ReceiverError;
+
+/// MPCOT receiver.
+#[derive(Debug, Default)]
+pub struct Receiver<T: state::State = state::Initialized> {
+    state: T,
+}
+
+impl Receiver {
+    /// Creates a new Receiver.
+    pub fn new() -> Self {
+        Receiver {
+            state: state::Initialized::default(),
+        }
+    }
+
+    /// Completes the setup phase of the protocol.
+    pub fn setup(self) -> Receiver<state::PreExtension> {
+        Receiver {
+            state: state::PreExtension { counter: 0 },
+        }
+    }
+}
+impl Receiver<state::PreExtension> {
+    /// Performs the prepare procedure in MPCOT extension.
+    /// Outputs the indices for SPCOT.
+    ///
+    /// # Arguments.
+    ///
+    /// * `alphas` - The queried indices.
+    /// * `n` - The total number of indices.
+    #[allow(clippy::type_complexity)]
+    pub fn pre_extend(
+        self,
+        alphas: &[u32],
+        n: u32,
+    ) -> Result<(Receiver<state::Extension>, Vec<(usize, u32)>), ReceiverError> {
+        let t = alphas.len() as u32;
+        if t > n {
+            return Err(ReceiverError::InvalidInput(
+                "the length of alpha should not exceed n".to_string(),
+            ));
+        }
+
+        // The range of each interval.
+        let k = (n + t - 1) / t;
+
+        let queries_length = if n % t == 0 {
+            vec![k as usize; t as usize]
+        } else {
+            let mut tmp = vec![k as usize; (t - 1) as usize];
```

Comment:
Hi, @xiangxiecrypto , I only just now got around to reviewing this PR.
Could you please clarify the following:

The paper says all intervals must be of the same size.
What is the justification of having the last interval be of a different size?
What if the last interval happens to be of size 1?
If there is a justification, could you formulate it, so I could add it here as a comment.
Or maybe we can just stick to always requiring that arguments ar such that `n % t == 0`?

Reply:
Actually, the last interval can't be 1 but it can be 2. My point above is still valid.
Consider if n == 1_000_001 and t == 1_000, the the last interval will be 2.

Reply:
In the current choosing parameters, we actually have `n % t == 0`. But in order to select more other parameters, I make it a little bit more flexible here.

Reply:
In the following implementations, I will add a more general way to estimate different LPN parameters. If  all the parameters we choose satisfy `n % t == 0`, I will remove the other part.